Stream_support/include/CGAL/IO/io.h

```diff
@@ -186,6 +188,30 @@ class Output_rep
   std::ostream& operator()( std::ostream& os) const { return (os << t); }
 };
 
+template <class T, class F>
+class Output_rep<std::optional<T>, F>
+{
+   const std::optional<T>& t;
+
+public:
+  Output_rep( const std::optional<T>& tt) : t(tt) {}
+  std::ostream& operator()( std::ostream& os) const {
+    if (t==std::nullopt) return (os << "--");
+    return (os << t.value()); }
+};
+
+template <class ... T, class F>
+class Output_rep<std::variant<T...>, F>
+{
+   const std::variant<T...>& t;
+
+public:
+  Output_rep( const std::variant<T...>& tt) : t(tt) {}
+  std::ostream& operator()( std::ostream& os) const {
+    return (os << "--");
+  }
+};
```

Comment:
Why print `--` and not the contained value? With `std::visit` and a generic lambda.

And indentation again:
```suggestion
template <class ... T, class F>
class Output_rep<std::variant<T...>, F>
{
  const std::variant<T...>& t;

public:
  Output_rep( const std::variant<T...>& tt) : t(tt) {}
  std::ostream& operator()( std::ostream& os) const {
    return std::visit([](auto&& v) {
                        return (os << v);
                      }, t);
  }
};
```

Note that I have not tested that code.

Reply:
That exactly the old behavior with boost. See https://github.com/boostorg/optional/blob/develop/include/boost/optional/optional_io.hpp

Reply:
> Note that I have not tested that code.

... at least the lambda has to catch `os` by reference.